syntax = "proto3";

package cat;

service CatService {
  // Notification que le port à été lu mais que la souris n'a pas été détectée
  rpc PortRead(MouseLocation) returns (Ack);
  // Notification d'une souris détectée par un autre chat
  rpc Notify(MouseLocation) returns (Ack);
  // Notification envoyée par Animal Services pour capturer le chat
  rpc CaptureNotification(CaptureRequest) returns (CaptureAck);
}

message MouseLocation {
  string node = 1;
  int32 port = 2;
}

message Ack {
  string message = 1;
}

message CaptureRequest {
  string cat_id = 1;
  int32 capture_duration = 2; // en secondes
}

message CaptureAck {
  string message = 1;
}
